ls1intum/Artemis
# [Bugfix] Fix Assessment Information Stats

### Checklist

### Motivation and Context


Solves issue #3890

### Description


The issue: It appears when a course/exam has a programming exercise, which has the assessment mode set to automatic, but has complaints enabled. (click the assessment type button *twice*)
![image](https://user-images.githubusercontent.com/33342534/128881763-1a4180b1-1872-4112-b087-2e67ed6052a6.png) -> ![image](https://user-images.githubusercontent.com/33342534/128881820-4776c844-02e4-4882-b36d-165dc8327f1f.png) -> ![image](https://user-images.githubusercontent.com/33342534/128881868-ec0c30aa-aa08-4cc2-9efa-27b93b382582.png)
When the hibernate query fetches the complaints for the assessment dashboard, it fails because the complaint is for an automatic result. Since the query groups by `result.assessor.id`, but for automatic results this is `null`.

I just investigated and[ found this stack](https://stackoverflow.com/questions/25322583/cant-instantiate-class-using-hibernate-createquery/29879294#29879294) overflow post. I replaced the primitive types used in the constructor (`long`, `double`) by `java.lang.Object` types (`Long`, `Double`). I also did this for the constructors used in the related classes `TutorLeaderboardComplaintResponses`, `TutorLeaderboardAssessments`, …

### Steps for Testing


1. Log in to Artemis as instructor
2. Create an exam at which you can participate as student afterwards, set a review start and end time for that exam. 
**2.1 add a programming exercise -> set assessment type to automatic, but with complaints enabled**
3. Participate as student and submit something
4. Navigate to the Assessment Dashboard of the corresponding exam as instructor
5. Check the stats shown in the information table on the top right side (as shown in the screenshot below)
6. If you, as instructor, retrieve the lock for the assessment by opening it, check that this information is updated in the table
7. If you, as instructor, submitted an assessment, check that this information updated in the table
8. As student, review the automatic assessment (only possible during review phase) and file a complaint
9. As an instructor, check that this information is reflected in the information table as well and that this complaint is linked in the table

### Test Coverage




### Screenshots


Having assessed the only submission for an exam and receiving a complaint about that assessment should lead to the following stats:
<img width="1235" alt="Bildschirmfoto 2021-08-09 um 21 28 05" src="https://user-images.githubusercontent.com/41366522/128763033-8564aae4-96b0-47aa-b935-2f2c8e08efe0.png">
